GenumKit/Parsers/FontsFileParser.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  FontsFileParser.swift
+//  Pods
+//
+//  Created by Derek Ostrander on 3/7/16.
+//
+//
+
+import Foundation
+import AppKit.NSFont
+
+typealias Font = (name: String, variant: String)
+
+extension CFString {
+    var alphanumericString: String {
+         return (self as String).componentsSeparatedByCharactersInSet(NSCharacterSet.alphanumericCharacterSet().invertedSet).joinWithSeparator("")
+    }
+}
+
+extension CTFont {
+    static func loadFont(path: String) -> Font? {
+        let inData = NSData(contentsOfFile: path)
+        var error: Unmanaged<CFError>?
+        guard
+            let provider: CGDataProvider = CGDataProviderCreateWithCFData(inData),
+            let font = CGFontCreateWithDataProvider(provider)
+            where CTFontManagerRegisterGraphicsFont(font, &error) else {
+                return nil
+        }
+        return CTFontCreateWithGraphicsFont(font, 0, nil, nil).font
+    }
+
+    private var font: Font? {
+        let family = CTFontCopyFamilyName(self).alphanumericString
+        let name = CTFontCopyFullName(self).alphanumericString.stringByReplacingOccurrencesOfString(family, withString: "")
+        if family.characters.count == 0 {
+            print("There was a problem detecting the name of the font. \(CTFontCopyFamilyName(self)) -- \(CTFontCopyFullName(self)) ")
+            return nil
+        }
+        return (family, name.characters.count > 0 ? name : "Regular")
+    }
+}
+
+// MARK: Interface
+public final class FontsFileParser {
+    public var entries: [String: Set<String>] = [:]
+
+    public init() {}
+
+    public func parseFonts(path: String) {
+        for file in contentsOfDirectory(path) {
+            let fullPath = (path as NSString).stringByAppendingPathComponent(file)
```

Comment:
Here's what I came up to in my tests to parse a directory, get the Universal Type Identifier of each file to only process files with an UTI conforming to "public.font", then parse only those files:

```
import Foundation
import MobileCoreServices
…
guard let dirEnum = NSFileManager.defaultManager().enumeratorAtURL(
  NSURL(fileURLWithPath: path),
  includingPropertiesForKeys:[NSURLTypeIdentifierKey],
  options: [.SkipsPackageDescendants, .SkipsHiddenFiles],
  errorHandler: nil) else {
    fatalError("Failed to enumerate directory \(path)")
}

var utiObject: AnyObject? = nil
while let file = dirEnum.nextObject() as? NSURL {
  // Retrieve UTI of the file
  guard let _ = try? file.getResourceValue(&utiObject, forKey: NSURLTypeIdentifierKey),
    let uti = utiObject as? String else {
      print("Unable to determine the Universal Type Identifier for file \(file)")
      continue
  }
  // Skip non-font files
  guard UTTypeConformsTo(uti, "public.font") else { continue }
  // Parse font file
  guard let font = CTFont.loadFont(fullPath) else { continue }
  addVariation(font.name, variation: font.variant)
}
```

HTH
---

PS: Once you've migrated to that solution, you will even be able to remove your private `contentsOfDirectory` and `isDirectory` functions below :+1: